Building 4.13.1.0 with GHC 8.6.5 fails

Trying to build with GHC 8.6.5 here, getting
```
lib/Hakyll/Core/Metadata.hs:69:23: error:
    Not in scope: type constructor or class ‘MonadFail’
   |
69 | getMetadataField' :: (MonadFail m, MonadMetadata m) => Identifier -> String -> m String
   |                       ^^^^^^^^^
```

#754 is probably related.

P.S. Probably just need to `import Control.Monad.Fail`?

--- a/lib/Hakyll/Core/Metadata.hs
+++ b/lib/Hakyll/Core/Metadata.hs
@@ -16,6 +16,7 @@ module Hakyll.Core.Metadata
 --------------------------------------------------------------------------------
 import           Control.Arrow                  (second)
 import           Control.Monad                  (forM)
+import           Control.Monad.Fail             (MonadFail)
 import           Data.Binary                    (Binary (..), getWord8,
                                                  putWord8, Get)
 import qualified Data.HashMap.Strict            as HMS

--- a/lib/Hakyll/Core/Rules/Internal.hs
+++ b/lib/Hakyll/Core/Rules/Internal.hs
@@ -13,6 +13,7 @@ module Hakyll.Core.Rules.Internal
 
 
 --------------------------------------------------------------------------------
+import           Control.Monad.Fail             (MonadFail)
 import           Control.Monad.Reader           (ask)
 import           Control.Monad.RWS              (RWST, runRWST)
 import           Control.Monad.Trans            (liftIO)

--- a/lib/Hakyll/Web/Template/Context.hs
+++ b/lib/Hakyll/Web/Template/Context.hs
@@ -54,6 +54,7 @@ module Hakyll.Web.Template.Context
 --------------------------------------------------------------------------------
 import           Control.Applicative           (Alternative (..))
 import           Control.Monad                 (msum)
+import           Control.Monad.Fail            (MonadFail)
 import           Data.List                     (intercalate, tails)
 #if MIN_VERSION_base(4,9,0)
 import           Data.Semigroup                (Semigroup (..))

--- a/lib/Hakyll/Web/Template/List.hs
+++ b/lib/Hakyll/Web/Template/List.hs
@@ -20,6 +20,7 @@ module Hakyll.Web.Template.List
 
 --------------------------------------------------------------------------------
 import           Control.Monad               (liftM)
+import           Control.Monad.Fail          (MonadFail)
 import           Data.List                   (intersperse, sortBy)
 import           Data.Ord                    (comparing)
 import           Data.Time.Locale.Compat     (defaultTimeLocale)

